src/inverted_index.c

```diff
@@ -1043,6 +1056,19 @@ int IR_Read(void *ctx, RSIndexResult **e) {
 
 #define BLOCK_MATCHES(blk, docId) ((blk).firstId <= docId && docId <= (blk).lastId)
 
+static int IndexReader_ReadWithSeeker(IndexReader *ir, t_docId* inOutDocId) {
```

Comment:
I don't like that we change the input soc id for the skip to function. This will cause future bugs